### DISCUSSION TITLE
Mux and ED parameter

### DISCUSSION THREAD

**Discussion by bart3nder** [XTLS/Xray-core#3612] [Q&A] [answered]
Hi! in transports like websocket and http upgrade, should i enable mux and ed=2560 at same time or one should be enough? i heard using both doesn’t make sense since they both do same thing.

**Comment by mmmray** [comment, marked as answer]
they both do similar things from user perspective ("reduce latency", "fix some traffic pattern"), but under the hood they don't do the same thing at all and can be used together to reduce latency even more.

* mux reduces latency when creating new connections, but during speedtest it can increase latency and/or reduce bandwidth actually
* ED just reduces latency, and there is no downside to enabling it.

**Reply by bart3nder** [reply]
Thanks!
